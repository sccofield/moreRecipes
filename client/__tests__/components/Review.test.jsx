import React from 'react';
import toJSON from 'enzyme-to-json';
import sinon from 'sinon';
import moxios from 'moxios';
import { Review, mapDispatchToProps, mapStateToProps }
  from '../../src/components/review/Review';

describe('Review controller', () => {
  beforeEach(() => {
    moxios.install();
  });
  afterEach(() => {
    moxios.uninstall();
  });

  const props = {
    addReviewActionCreator: jest.fn(() => Promise.resolve()),
    reviews: {
      userName: 'mike',
      createdAt: 'today',
      review: 'review'
    },
    recipeId: 2,
    isAuthenticated: true

  };


  it('should render', () => {
    const wrapper = shallow(<Review {...props} />);
    expect(wrapper.find('.review').exists).toBeTruthy();
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it(
    'sets the state when the value of the input changes',
    () => {
      const spy = sinon.spy(Review.prototype, 'onChange');
      const wrapper = shallow(<Review {...props} />);
      const event = {
        target: {
          value: 'nice review',
          name: 'review'
        }
      };
      wrapper.instance().onChange(event);
      expect(wrapper.state('review')).toBe('nice review');

      expect(spy.called).toBeTruthy();
    }
  );
});

describe('conatiner functions', () => {
  test('mapStateToProps', () => {
    expect(mapStateToProps({
      recipe: {
        singleRecipe: {
          Reviews: 'reviews'
        }
      },
      user: {
        isAuthenticated: true
      }
    })).toHaveProperty('reviews');
  });

  test('mapDispatchToProps', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch))
      .toHaveProperty('addReviewActionCreator');

    const { addReviewActionCreator } = mapDispatchToProps(dispatch);
    addReviewActionCreator();
    expect(dispatch).toHaveBeenCalled();
  });
});

