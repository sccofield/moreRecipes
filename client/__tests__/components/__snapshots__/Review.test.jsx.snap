// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review controller should render 1`] = `
<div
  className=""
>
  <h3
    className="text-center"
  >
    Reviews
  </h3>
  <hr />
  <div
    className="review"
  >
    <form
      className="form-group md-form"
      onChange={[Function]}
      onSubmit={[Function]}
    >
      <textarea
        className="form-control"
        id="exampleTextarea"
        name="review"
        placeholder="Enter review here"
        rows="4"
        value=""
      />
      <button
        className="btn btn-primary recipeButton mt-2"
        type="submit"
      >
        Add Review
      </button>
    </form>
  </div>
</div>
`;
